File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int ops = 0;

        for(int n : nums) {
            if(n < k) pq.add(n);
        }

        while(!pq.isEmpty()) {
            int x = pq.poll();
            ops++;

            if(pq.isEmpty()) break;
            int y = pq.poll();

            long val = 2L * x + y;
            if(val < k) pq.add((int)val);
        }

        return ops;
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for(auto it : nums) {
            if(it < k) pq.push(it);
        }

        int ops = 0;
        while(!pq.empty()) {
            int x = pq.top();
            pq.pop();
            ops++;

            if(pq.empty()) break;
            int y = pq.top();
            pq.pop();

            long long val = 2L * x + y;
            if(val < k) pq.push((int)val);
        }

        return ops;
    }
};